homeassistant/components/advantage_air/climate.py

```diff
@@ -174,7 +162,12 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set the Fan Mode."""
-        await self.async_update_ac({"fan": HASS_FAN_MODES.get(fan_mode)})
+        mode = (
+            ADVANTAGE_AIR_MYFAN
```

Comment:
This looks to me like it should raise instead to alert the user he can't control the fan when it's on auto.
Just setting something the user isn't asking for seems wrong.

Reply:
You're right that this logic is wrong. The user should be able to change the fan mode. I will fix.